src/containers/Modal/index.jsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import AlertModal from './Alert'
+import FormModal from './Form'
+import FullScreenModal from './FullScreen'
+import MaxWidthModal from './MaxWidth'
+import { ButtonsContainer } from './style'
+
+function Modal() {
+  const [opened, setOpened] = React.useState('')
+
+  const handleClose = () => setOpened('')
+
+  const openModal = value => () => setOpened(value)
+  
+  return (
+    <ButtonsContainer container>
+      <AlertModal open={opened} openModal={openModal} handleClose={handleClose} />
```

Comment:
O controle do open poderia ficar aqui:
```
<AlertModal open={opened === 'alert'} />
```

Por que assim, a regra de abrir e fechar você totalmente nesse container e seu componente de alert fica reativo... Dessa forma seus componentes de modal estão "amarrados" com a regra de abrir o modal.

Se for olhar mais ainda pro lado de deixar mais generico e dividir responsabilidades, os botões de abrir deveriam ficar todos nesse componente, pois assim você teria apenas a prop open e evitaria a prop openModal e deixaria mais componetizado seus modal